agent/bench-scripts/postprocess/process-iteration-samples

```diff
@@ -226,18 +233,27 @@ foreach $sample (@json_samples) { # samples 0..N
 							if ( not defined $iteration{$metric_type}{$metric_name}[$iteration_matching_index]{get_label('samples_label')} ) {
 								# Create an array to hold the samples for "value" values
 								# For example, if uperf had 5 sample runs, there would be an array of 5 Gb_sec values in this array
-								#my @samples;
 								$iteration{$metric_type}{$metric_name}[$iteration_matching_index]{get_label('samples_label')} = ();
 							}
 							push @{$iteration{$metric_type}{$metric_name}[$iteration_matching_index]{get_label('samples_label')}}, \%samples_hash;
-						# For the rest of the keys, either check to see if it matches existing value in the %iteration, 
-						# or if it does not exist, create it.
 						} else {
+							# For the rest of the keys, either check to see if it matches existing value in the %iteration,
+							# or if it does not exist, create it.
 							if ( $key ne get_label('timeseries_label') and $key ne get_label('uid_label') ) { # timeseries already handled above
 								if ($iteration{$metric_type}{$metric_name}[$iteration_matching_index]{$key}) {
```

Comment:
Should be using `defined` here.

Reply:
After reading a bit, it seems like this should really be `exists()` instead of `defined()`, both here and in other places.